src/main/java/com/org/jenkins/custom/jenkins/distribution/service/generators/WarGenerator.java

```diff
@@ -0,0 +1,29 @@
+package com.org.jenkins.custom.jenkins.distribution.service.generators;
+
+import com.org.jenkins.custom.jenkins.distribution.service.util.Util;
+import io.jenkins.tools.warpackager.lib.config.Config;
+import io.jenkins.tools.warpackager.lib.impl.Builder;
+
+import java.io.File;
+
+public class WarGenerator {
+
+    private static Util util = new Util();
+
+    private static final String DEFAULT_TMP_DIR_NAME = "tmp";
+    private static final String PACKAGER_CONFIG_YAML = "packager-config.yml";
+
+    public static void generateWAR(String versionName) {
+        final Config cfg;
+        final File configPath = util.getFileFromResources(PACKAGER_CONFIG_YAML);
+        try {
+            cfg = Config.loadConfig(configPath);
+            cfg.buildSettings.setTmpDir(new File(DEFAULT_TMP_DIR_NAME));
```

Comment:
You can get a temporary directory by the following code:
`System.getProperty("java.io.tmpdir")`
Furthermore, a user can give it a different value by passing a path. For example: `java -Djava.io.tmpdir=/path/to/tmpdir`

Reply:
Yes made the change :+1: